Remove !HAVE(ALTERNATE_ICONS) paths

hortont424 opened the pull request:
#### 55a6cdba7ec28fcc7759670bed4fa4ef5ea44efc
<pre>
Remove !HAVE(ALTERNATE_ICONS) paths
<a href="https://bugs.webkit.org/show_bug.cgi?id=291016">https://bugs.webkit.org/show_bug.cgi?id=291016</a>
<a href="https://rdar.apple.com/148541433">rdar://148541433</a>

Reviewed by Megan Gardner and Abrar Rahman Protyasha.

Remove the non-HAVE(ALTERNATE_ICONS) paths; always use the SF Symbol-derived icons.
Remove the old AttachmentPlaceholder bitmaps.
Adopt the public API for symbol image construction and style adjustment.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/mac/NSImageSPI.h:
* Source/WebCore/Resources/AttachmentPlaceholder.png: Removed.
* Source/WebCore/Resources/AttachmentPlaceholder@2x.png: Removed.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/html.css:
(input::-webkit-contacts-auto-fill-button):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::createAttachmentPlaceholderImage):

Canonical link: <a href="https://commits.webkit.org/293202@main">https://commits.webkit.org/293202@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e74e24846b238e0c3ea40ed14dbdc0ebca23773f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98181 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17812 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8039 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103298 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/48710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100226 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18104 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26263 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/74746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/48710 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13722 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/88709 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/55106 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13504 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6655 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/48152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/90862 "Failed to checkout and rebase branch from PR 43563") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/83477 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6734 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105674 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/96807 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25267 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/18404 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/83733 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25640 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84899 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83192 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21012 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27832 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/5520 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/18907 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25226 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/30400 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/120428 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25046 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/33737 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/28362 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26621 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1474,28 +1474,14 @@ static void paintAttachmentIcon(const RenderAttachment& attachment, GraphicsCont
-        NSImageHintSymbolScale : @(NSImageSymbolScaleMedium)
-ALLOW_DEPRECATED_DECLARATIONS_END
-    }];
+    auto cgImage = [image CGImageForProposedRect:&imageRect context:nil hints:nil];
```
```suggestion
    RetainPtr cgImage = [image CGImageForProposedRect:&imageRect context:nil hints:nil];
```

hortont424 replied:
heh, I caught the other two but not this one. good call.

hortont424 committed:
Remove !HAVE(ALTERNATE_ICONS) paths
https://bugs.webkit.org/show_bug.cgi?id=291016
rdar://148541433

Reviewed by Megan Gardner and Abrar Rahman Protyasha.

Remove the non-HAVE(ALTERNATE_ICONS) paths; always use the SF Symbol-derived icons.
Remove the old AttachmentPlaceholder bitmaps.
Adopt the public API for symbol image construction and style adjustment.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/mac/NSImageSPI.h:
* Source/WebCore/Resources/AttachmentPlaceholder.png: Removed.
* Source/WebCore/Resources/AttachmentPlaceholder@2x.png: Removed.
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/html.css:
(input::-webkit-contacts-auto-fill-button):
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
(WebCore::createAttachmentPlaceholderImage):

Canonical link: https://commits.webkit.org/293202@main
